Sources/bip32-ed25519-swift/bip32-ed25519-swift.swift

```diff
@@ -167,7 +176,43 @@ public class Bip32Ed25519 {
         return (z, childChainCode)
     }
 
-    func deriveChildNodePrivate(extendedKey: Data, index: UInt32) -> Data {
+    func deriveChildNodePublic(extendedKey: Data, keyIndex: UInt32, g: Int = 9) throws -> Data {
+        guard keyIndex < 0x8000_0000 else {
+            throw DataValidationException(message: "Cannot derive public key with harden")
+        }
+
+        let pk = extendedKey.subdata(in: 0 ..< 32)
+        let cc = extendedKey.subdata(in: 32 ..< 64)
```

Comment:
for the CC parts maybe we can have CHAIN_CODE_SIZE = 32

Reply:
if the chain code is always at those indices could we just do something like `let CHAIN_CODE_INDICES = 32 ..< 64`?